src/compiler/scala/tools/reflect/FastStringInterpolator.scala

```diff
@@ -57,40 +59,20 @@ trait FastStringInterpolator extends FormatInterpolator {
           case iue: StringContext.InvalidUnicodeEscapeException => c.abort(parts.head.pos.withShift(iue.index), iue.getMessage)
         }
 
-      val argsIndexed = args.toVector
-      val concatArgs = collection.mutable.ListBuffer[Tree]()
-      val numLits = parts.length
-      foreachWithIndex(treated.tail) { (lit, i) =>
-        val treatedContents = lit.asInstanceOf[Literal].value.stringValue
-        val emptyLit = treatedContents.isEmpty
-        if (i < numLits - 1) {
-          concatArgs += argsIndexed(i)
-          if (!emptyLit) concatArgs += lit
-        } else if (!emptyLit) {
-          concatArgs += lit
+      if (args.forall(cond(_) { case Literal(c @ Constant(_)) => c.isSuitableLiteralType })) {
```

Comment:
We should exclude `Unit`, `Float` and `Double` here, because they do not have a portable `toString()` representation. If we convert them to strings in the compiler, that will always use the JVM `toString()`, so the result won't be correct on other platforms.

Reply:
It's like the embedded line ending in multiline strings, except there is no way to request it; it would need a different interpolator.